### PR TITLE
Describe measurement documents

### PR DESCRIPTION
**Pull request by wfleming**

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -323,6 +328,20 @@ An example trace:
 
 ```
 
+### Measurements
+
+Measurements are a separate kind of document that engines can emit. Measurements
+are a numeric value to a given named key. Measurements are primarily used for
+tracking rates of certain kinds of errors engines may encounter internally (e.g.
+parse errors).
```
Comment (dblandin) [inline SPEC.md]: What do you think of generalizing this description a bit?

Maybe something like...

> Measurements are a numeric value bound to a named key. Measurements may represent a variety of internal engine metrics. For example, an engine may emit a measurement representing the rate of parsing errors during an analysis.

File: SPEC.md [hunk 2]
```diff
@@ -323,6 +328,20 @@ An example trace:
...
+
+```json
+{
+  "type": "measurement",
+  "name": "foo.bar",
+  "value": 42
```
Comment (dblandin) [inline SPEC.md]: Are we limiting value to integers or will we allow floats as well?

Comment (wfleming) [reply SPEC.md]: Given the conversation with Bryan & the potential larger future purposes for this document type, I'm not sure it's reasonable to limit ourselves to ints at the spec level.

File: SPEC.md (hunk 1 above)
Comment (wfleming) [reply SPEC.md]: Updated. These documents are not always going to be for "internal engine metrics", though: that's just our first intended use of them. I rephrased that sentence a bit.

File: SPEC.md (hunk 2 above)
Comment (dblandin) [reply SPEC.md]: Sounds good. Just wasn't clear from the spec or example if floats were allowed as well.

File: SPEC.md [hunk 3]
```diff
@@ -323,6 +328,20 @@ An example trace:
 
 ```
 
+### Measurements
+
+Measurements are a separate kind of document that engines can emit. Measurements
+are a numeric value bound to a named key. One common use of measurements is
+representing of internal engine metrics. For example, an engine may emit a
```
Comment (dblandin) [inline SPEC.md]: Should we ✂️ this second "of"? -> representing internal engine metrics

Comment (wfleming) [reply SPEC.md]: Woops, thanks.

**Comment by wfleming** [comment]
Going to leave this open & not merge for a while. This is an early idea we're working with & may have internal/beta implementation support for soon, but aren't going to consider a stable part of the spec yet.

**Comment by wfleming** [comment]
Closing. We are indeed using this schema privately but have not seen any demand for this to be more widely available & have no plans to support it more widely any time soon.

**Comment by emircan2134** [comment]
Duplicate of #

**Pull request closed without merging**